src/math/CurveFitting.cc

```diff
@@ -123,75 +119,53 @@ namespace pfs{ namespace drp{ namespace stella{ namespace math{
-    PTR(ndarray::Array<T, 1, 1>) P_D_A1_MeasureErrors(new ndarray::Array<T, 1, 1>(measureErrors));
+    ndarray::Array<T, 1, 1> D_A1_MeasureErrors = ndarray::allocate(D_A1_X_In.getShape()[0]);
+    D_A1_MeasureErrors.deep() = 1.;
+    PTR(ndarray::Array<T, 1, 1>) P_D_A1_MeasureErrors(new ndarray::Array<T, 1, 1>(D_A1_MeasureErrors));
 
     int I_Pos = -1;
-    I_Pos = pfs::drp::stella::utils::KeyWord_Set(S_A1_Args_In, "MEASURE_ERRORS");
-    if (I_Pos >= 0){
+    if ((I_Pos = pfs::drp::stella::utils::KeyWord_Set(S_A1_Args_In, "MEASURE_ERRORS")) >= 0){
+      LOGLS_DEBUG(_log, "Reading MEASURE_ERRORS");
       P_D_A1_MeasureErrors.reset();
       P_D_A1_MeasureErrors = (*((PTR(ndarray::Array<T, 1, 1>)*)ArgV[I_Pos]));
-      measureErrors.deep() = *P_D_A1_MeasureErrors;
       B_HaveMeasureErrors = true;
       if (P_D_A1_MeasureErrors->getShape()[0] != D_A1_X_In.getShape()[0]){
         string message("pfs::drp::stella::math::CurfFitting::PolyFit: Error: P_D_A1_MeasureErrors->getShape()[0](=");
         message += to_string(P_D_A1_MeasureErrors->getShape()[0]) + ") != D_A1_X_In.getShape()[0](=" + to_string(D_A1_X_In.getShape()[0]) + ")";
-        cout << message << endl;
         throw LSST_EXCEPT(pexExcept::Exception, message.c_str());
       }
     }
-    else{
-      for (auto it = measureErrors.begin(); it != measureErrors.end(); ++it)
-        *it = (*it > 0) ? sqrt(*it) : 1;
-    }
-    ndarray::Array<T, 1, 1> D_A1_MeasureErrorsBak = copy(measureErrors);
 
     PTR(std::vector<size_t>) P_I_A1_NotRejected(new std::vector<size_t>());
     I_Pos = pfs::drp::stella::utils::KeyWord_Set(S_A1_Args_In, "NOT_REJECTED");
     if (I_Pos >= 0){
       P_I_A1_NotRejected.reset();
       P_I_A1_NotRejected = *((PTR(std::vector<size_t>)*)(ArgV[I_Pos]));
-      #ifdef __DEBUG_POLYFIT__
-        cout << "PolyFit: P_I_A1_NotRejected = ";
-        for (auto it = P_I_A1_NotRejected->begin(); it != P_I_A1_NotRejected->end(); ++it)
-          cout << *it << " ";
-        cout << endl;
-        cout << "pfs::drp::stella::math::CurveFitting::PolyFit: KeyWord NOT_REJECTED read" << endl;
-      #endif
+      LOGLS_DEBUG(_log, "KeyWord NOT_REJECTED read");
     }
+    P_I_A1_NotRejected->resize(0);
 
     PTR(std::vector<size_t>) P_I_A1_Rejected(new std::vector<size_t>());
     I_Pos = pfs::drp::stella::utils::KeyWord_Set(S_A1_Args_In, "REJECTED");
     if (I_Pos >= 0){
       P_I_A1_Rejected.reset();
       P_I_A1_Rejected = *((PTR(std::vector<size_t>)*)(ArgV[I_Pos]));
-      #ifdef __DEBUG_POLYFIT__
-        cout << "PolyFit: P_I_A1_NotRejected = ";
-        for (auto it = P_I_A1_Rejected->begin(); it != P_I_A1_Rejected->end(); ++it)
-          cout << *it << " ";
-        cout << endl;
-        cout << "pfs::drp::stella::math::CurveFitting::PolyFit: KeyWord REJECTED read" << endl;
-      #endif
+      LOGLS_DEBUG(_log, "KeyWord REJECTED read");
     }
+    P_I_A1_Rejected->resize(0);
```

Comment:
Why has it only become necessary to initialize this (and ``P_I_A1_NotRejected``, ``P_I_NRejected``) now? Is it purely cosmetic, or is this fixing a bug?

If it's fixing a bug, please add a test case.